supervisor/dbus/udisks2/block.py

```diff
@@ -0,0 +1,244 @@
+
+    @property
+    @dbus_property
+    def id(self) -> str:
+        """Return unique identifer."""
+        return self.properties[DBUS_ATTR_ID]
+
+    @property
+    @dbus_property
+    def read_only(self) -> bool:
+        """Return whether device is read only."""
+        return self.properties[DBUS_ATTR_READ_ONLY]
+
+    @property
+    @dbus_property
+    def symlinks(self) -> list[str]:
+        """Return list of symlinks."""
+        return [
+            bytes(symlink).decode() for symlink in self.properties[DBUS_ATTR_SYMLINKS]
+        ]
+
+    @property
+    @dbus_property
+    def device_number(self) -> int:
+        """Return device number."""
+        return self.properties[DBUS_ATTR_DEVICE_NUMBER]
+
+    @property
+    @dbus_property
+    def configuration(
+        self,
+    ) -> dict[str, list[FstabConfigDetails | CrypttabConfigDetails]]:
+        """Return device configuration."""
+        configuration = defaultdict(list)
+        for type_, details in self.properties[DBUS_ATTR_CONFIGURATION]:
+            configuration[type_].append(
+                FstabConfigDetails.from_dict(details)
+                if "dir" in details
+                else CrypttabConfigDetails.from_dict(details)
+            )
+        return dict(configuration)
+
+    @property
+    @dbus_property
+    def size(self) -> int:
+        """Return size."""
+        return self.properties[DBUS_ATTR_SIZE]
+
+    @dbus_connected
+    async def add_configuration_item(
+        self,
+        type_: str,
+        details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Add new configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.AddConfigurationItem(
+            ("(sa{sv})", (type_, details.to_dict())), ("a{sv}", options)
+        )
+        await self.update()
+
+    @dbus_connected
+    async def remove_configuration_item(
+        self,
+        type_: str,
+        details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Remove existing configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.RemoveConfigurationItem(
+            ("(sa{sv})", (type_, details.to_dict())), ("a{sv}", options)
+        )
+        await self.update()
+
+    @dbus_connected
+    async def update_configuration_item(
+        self,
+        old_type: str,
+        old_details: FstabConfigDetails | CrypttabConfigDetails,
+        new_type: str,
+        new_details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Add new configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.AddConfigurationItem(
+            ("(sa{sv})", (old_type, old_details.to_dict())),
+            ("(sa{sv})", (new_type, new_details.to_dict())),
+            ("a{sv}", options),
+        )
```

Comment:
```suggestion
        await self.dbus.Block.call_add_configuration_item(
            (old_type, old_details.to_dict()),
            (new_type, new_details.to_dict()),
            options,
        )
```

Reply:
Also realized this should be `call_update_configuration_item`. Was very confused at the two different signatures, had never seen that before.

Reply:
Suggested changes are accounted for. Leaving open because we need to adjust the dictionaries to have a `Variant` as the value for each key.